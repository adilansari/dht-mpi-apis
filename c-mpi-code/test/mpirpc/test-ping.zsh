#!/bin/zsh

# Be sure to make tests with D=1

source tools/test-helpers.zsh

# set -x

PROGRAM=$1
OUTPUT=${PROGRAM%.x}.out
shift
LAUNCH=${*}

eval ${LAUNCH} -n 2 ${PROGRAM} > ${OUTPUT} 2>&1
CODE=$?
[[ ${CODE} == 0 ]] || crash "exit code was: ${CODE}"

# Should be 5 calls to handle_ping()...
N=$( grep -c "handle_ping" ${OUTPUT} )
(( N == 5 )) || crash "N != 6"

grep "handle_quit" ${OUTPUT} > /dev/null || crash "No handle_quit!"

# Should be 2 "Normal exit."s
N=$( grep -c "Normal exit." ${OUTPUT} )
(( N == 2 )) || crash "N != 2"

exit 0
