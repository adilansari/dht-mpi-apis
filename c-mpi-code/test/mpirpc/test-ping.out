hello
hello
Debug to stdout
[1] debug_rank: 1
[0] debug_rank: 0
handle_ping
handle_ping
handle_ping
handle_ping
handle_ping
handle_quit
[0] Normal exit.
 
[1] Normal exit.
 
